src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -37,6 +42,81 @@ const ChannelListScreenV2 = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  React.useEffect(() => {
+    const checkNotif = async () => {
+      if (!navigation.isFocused()) return;
+      PushNotification.checkPermissions(
+        (
+          data: PushNotificationPermissions & {
+            authorizationStatus?: number;
+          }
+        ) => {
+          const lastPromptTime = StorageUtils.lastPromptNotification.get();
+
+          // For testing purpose,  (20000 milliseconds)
+          const promptInterval = 20000;
+          const currentTime = new Date().getTime();
+          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+            return; // Don't show the prompt if the interval has not passed
+          }
+
+          const needsPermission =
+            data.alert === false || data.badge === false || data.sound === false;
+
+          const showAlert = (onPressAction: () => void) =>
+            Alert.alert(
+              "Don't Miss New Message",
+              'Allow notification to know when friends send you messages',
+              [
+                {text: 'Not now', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
+                {text: 'Allow', onPress: onPressAction}
+              ]
+            );
+
+          switch (data.authorizationStatus) {
+            case 1:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              } else {
+                requestPermission();
+              }
+              break;
+            case 2:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              }
+              break;
+            case 3:
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => requestPermission());
+              break;
+            default:
+              break;
+          }
+        }
+      );
+    };
+
+    checkNotif();
+  }, [navigation?.isFocused?.()]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `useEffect` hook, you're checking notification permissions and prompting the user if needed. However, there are some issues with this implementation:

1. The `checkNotif` function is defined inside the `useEffect` hook, which means it will be redefined every time the component renders. This could lead to performance issues. Consider moving the function definition outside of the `useEffect` hook.

2. There's no error handling for the `PushNotification.checkPermissions` call. If this operation fails, the app might crash. Wrap this call in a try-catch block to handle potential errors.

3. The `showAlert` function is defined inside the callback passed to `PushNotification.checkPermissions`. This function doesn't depend on any variables from the callback, so it can be moved outside for better readability and performance.

4. The `console.log` and `console.warn` statements used in the `Alert` buttons and `openSettings` catch block are not very informative. Consider replacing them with more descriptive messages or handling the errors in a more user-friendly way.

Here's how you could improve this code:

```diff
+ const showAlert = (onPressAction: () => void) =>
+   Alert.alert(
+     "Don't Miss New Message",
+     'Allow notification to know when friends send you messages',
+     [
+       {text: 'Not now', onPress: () => console.log('Notification permission prompt cancelled'), style: 'cancel'},
+       {text: 'Allow', onPress: onPressAction}
+     ]
+   );

+ const checkNotif = async () => {
+   if (!navigation.isFocused()) return;
+   try {
+     PushNotification.checkPermissions(
+       (
+         data: PushNotificationPermissions & {
+           authorizationStatus?: number;
+         }
+       ) => {
+         const lastPromptTime = StorageUtils.lastPromptNotification.get();

+         // For testing purpose,  (20000 milliseconds)
+         const promptInterval = 20000;
+         const currentTime = new Date().getTime();
+         if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+           return; // Don't show the prompt if the interval has not passed
+         }

+         const needsPermission =
+           data.alert === false || data.badge === false || data.sound === false;

+         switch (data.authorizationStatus) {
+           case 1:
+             if (needsPermission) {
+               StorageUtils.lastPromptNotification.set(currentTime.toString());
+               showAlert(() => openSettings().catch(() => console.warn('Cannot open settings')));
+             } else {
+               requestPermission();
+             }
+             break;
+           case 2:
+             if (needsPermission) {
+               StorageUtils.lastPromptNotification.set(currentTime.toString());
+               showAlert(() => openSettings().catch(() => console.warn('Cannot open settings')));
+             }
+             break;
+           case 3:
+             StorageUtils.lastPromptNotification.set(currentTime.toString());
+             showAlert(() => requestPermission());
+             break;
+           default:
+             break;
+         }
+       }
+     );
+   } catch (error) {
+     console.error('Error while checking permissions:', error);
+   }
+ };

- React.useEffect(() => {
-   const checkNotif = async () => {
-     if (!navigation.isFocused()) return;
-     PushNotification.checkPermissions(
-       (
-         data: PushNotificationPermissions & {
-           authorizationStatus?: number;
-         }
-       ) => {
-         const lastPromptTime = StorageUtils.lastPromptNotification.get();

-         // For testing purpose,  (20000 milliseconds)
-         const promptInterval = 20000;
-         const currentTime = new Date().getTime();
-         if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
-           return; // Don't show the prompt if the interval has not passed
-         }

-         const needsPermission =
-           data.alert === false || data.badge === false || data.sound === false;

-         const showAlert = (onPressAction: () => void) =>
-           Alert.alert(
-             "Don't Miss New Message",
-             'Allow notification to know when friends send you messages',
-             [
-               {text: 'Not now', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
-               {text: 'Allow', onPress: onPressAction}
-             ]
-           );

-         switch (data.authorizationStatus) {
-           case 1:
-             if (needsPermission) {
-               StorageUtils.lastPromptNotification.set(currentTime.toString());
-               showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
-             } else {
-               requestPermission();
-             }
-             break;
-           case 2:
-             if (needsPermission) {
-               StorageUtils.lastPromptNotification.set(currentTime.toString());
-               showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
-             }
-             break;
-           case 3:
-             StorageUtils.lastPromptNotification.set(currentTime.toString());
-             showAlert(() => requestPermission());
-             break;
-           default:
-             break;
-         }
-       }
-     );
-   };

-   checkNotif();
- }, [navigation?.isFocused?.()]);
+ React.useEffect(() => {
+   checkNotif();
+ }, [navigation?.isFocused?.()]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->